src/BugsnagUnity/Delivery.cs

```diff
@@ -524,21 +524,54 @@ private IEnumerator ProcessCachedItems(Type t)
 
                     try
                     {
+                        // if something goes wrong at this stage then we silently discard the file as it's most likley an io error at some point
                         payloadDictionary = ((JsonObject)SimpleJson.DeserializeObject(payloadJson)).GetDictionary();
                     }
                     catch (Exception e)
                     {
-                        Debug.LogException(new Exception("Bugsnag Error. Deserialising a cached payload for delivery failed: " + e.Message + " Cached json: " + payloadJson));
+                        _cacheManager.RemoveCachedEvent(id);
                         continue;
                     }
 
+
+                    // if something goes wrong at this stage then we discard the file and report the error as it might be a bug in the sdk
                     if (isSession)
                     {
-                        Deliver(new SessionReport(_configuration, payloadDictionary));
+                        try
+                        {
+                            var sessionReport = new SessionReport(_configuration, payloadDictionary);
+                            Deliver(sessionReport);
+                        }
+                        catch (Exception e)
+                        {
+                            Bugsnag.Notify(e,(@event)=>
+                            {
+                                @event.Context = "BugSnag Session Cache Error";
+                                @event.AddMetadata("BugsnagSessionData",payloadDictionary);
```

Comment:
This feels like we're skirting around internal error reports – adding a new tab to a customer's event isn't something I'm aware of us doing elsewhere for notifier issues.

@richardelms – can you take a bit of a look at the Android and Cocoa implementations for Internal Events so you can see how we achieve it and how much work it would be to adopt this mechanism here? I believe it's used for almost exactly this scenario: serialisation/deserialisation issues.